src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -31,20 +32,25 @@
  * Implementation of Query interface for type "hybrid". It allows execution of multiple sub-queries and collect individual
  * scores for each sub-query.
  */
+@Getter
 public final class HybridQuery extends Query implements Iterable<Query> {
 
     private final List<Query> subQueries;
+    private int paginationDepth;
```

Comment:
this parameter is low level for the Query level, can you create some more generic DTO for params like this, some sort of QueryContext, where pagination will be one of the parameters.We can put `explain` and `sort` there too, then pass only that context between the Query-QueryBuilder and even combination/normalization related classes

Reply:
I can create a HybridQueryContext dto object. Also I will add pagination depth in it. However, I will avoid touching explain and sort in this PR

Reply:
Makes sense, let add the class for context and with field for pagination, we can refactor other fields in later PR